File: AprovaFacil.Client/src/types/request.ts
import { UserResponse } from "./auth";
import { CompanyResponse } from "./company";

export interface RequestReponse {
  uuid: string;
  requester_id: number;
  invoice_name: string;
  budget_name: string;
  has_invoice: boolean;
  has_budget: boolean;
  payment_date: string;
  create_at: string;
  first_level_at: string;
  approved_first_level: number;
  second_level_at: string;
  approved_second_level: number;
  approved: number;
  approved_label: string;
  level: number;
  received_at: string;
  amount: number;
  note: string;
  company: CompanyResponse;
  requester: UserResponse;
  finisher: UserResponse;
  managers: UserResponse[];
  directors: UserResponse[];
}

export interface MonthlyStatusData {
  month: string
  pending: number
  approved: number
  rejected: number
}

export interface RequestStatsResponse {
  total_requests: number
  total_request_pending: number
  total_request_approved: number
  total_request_rejected: number
  total_amount_requests_approved: number
  total_requests_by_month: MonthlyStatusData[] // atualizado!
}

File: AprovaFacil.Domain/Models/Request.cs
﻿namespace AprovaFacil.Domain.Models;

public class Request
{
    public Guid UUID { get; set; }

    public Int32 CompanyId { get; set; }
    public Int32 RequesterId { get; set; }
    public Int32? FinisherId { get; set; }

    public Guid InvoiceName { get; set; }
    public Boolean HasInvoice { get; set; }
    public Guid BudgetName { get; set; }
    public Boolean HasBudget { get; set; }

    public DateTime? PaymentDate { get; set; }
    public DateTime CreateAt { get; set; }

    public DateTime? FirstLevelAt { get; set; }
    public Boolean ApprovedFirstLevel { get; set; }
    public DateTime? SecondLevelAt { get; set; }
    public Boolean ApprovedSecondLevel { get; set; }

    public Int32 Level { get; set; }
    public Int32 Status { get; set; }

    public DateTime? FinishedAt { get; set; }

    public Int64 Amount { get; set; }
    public String? Note { get; set; }

    public Int32 TenantId { get; set; }
    public Tenant Tenant { get; set; } = null!;

    public Company Company { get; set; } = null!;

    public IApplicationUser Requester { get; set; } = null!;
    public IApplicationUser? Finisher { get; set; }

    public List<RequestManager> Managers { get; set; } = new();
    public List<RequestDirector> Directors { get; set; } = new();

    public List<Notification> Notifications { get; set; } = new();
}

public class RequestManager
{
    public Guid RequestUUID { get; set; }
    public Int32 ManagerId { get; set; }

    public Int32 Approved { get; set; }

    public Request Request { get; set; } = null!;
    public IApplicationUser User { get; set; } = null!;
}

public class RequestDirector
{
    public Guid RequestUUID { get; set; }
    public Int32 DirectorId { get; set; }

    public Int32 Approved { get; set; }

    public Request Request { get; set; } = null!;
    public IApplicationUser User { get; set; } = null!;
}